File: 0929-unique-email-addresses/0929-unique-email-addresses.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        int n = emails.size();

        for (int i = 0; i < n; i++) {

            for (int j = 0; j < emails[i].size(); j++) {

                if (emails[i][j] == '@')
                    break;

                if (emails[i][j] == '.') {
                    emails[i].erase(emails[i].begin() + j);
                    j--;
                }

                if (emails[i][j] == '+') {
                    int at = emails[i].find('@');
                    emails[i].erase(emails[i].begin() + j,
                                    emails[i].begin() + at);
                    break;
                }
            }
        }

        set<string> s(emails.begin(), emails.end());

        return s.size();
    }
};

File: 0929-unique-email-addresses/0929-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        n = len(emails)

        for i in range(n):
            emails[i] = list(emails[i])

            j = 0
            while j < len(emails[i]):

                if emails[i][j] == '@':
                    break

                if emails[i][j] == '.':
                    emails[i].remove('.')
                    continue

                if emails[i][j] == '+':
                    at = emails[i].index('@')
                    del emails[i][j:at]
                    break

                j += 1

            emails[i] = ''.join(emails[i])

        return len(set(emails))